### PR TITLE
feat: support option to omit path in error messages

### PR DESCRIPTION
**Pull request by thanoskrg**
This PR adds an extra optional property `includePath?: boolean` under `type FromZodIssueOptions`.

With that option we allow users of `fromZodIssue` or `fromZodError` methods to take control on whether to include the `"at {path}"` suffix in their error messages. 

To ensure backwards compatibility, this option is optional and by default set to `true`. 

Resolves https://github.com/causaly/zod-validation-error/issues/134

### CODE REVIEW

**Review by nikoskalogridis — APPROVED** [review]
lgtm

File: package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
     "jest": "^29.3.1",
     "lint-staged": "^15.0.1",
     "prettier": "^3.0.2",
+    "prettier-2": "npm:prettier@^2.8.8",
```
Comment (jmike) [inline package.json:79]: question:

1. Why can't we user the current version of prettier?
2. What does the `npm:` prefix do?

Comment (thanoskrg) [reply package.json:79]: I have added relevant reference here:

https://github.com/causaly/zod-validation-error/blob/f46e7fa917383e9ccff511f17d281390cd42a781/jest.config.js#L5-L7

I hope this explains that. 
The root cause that the new tests were failing, because I couldn't get inline snapshots to work with the prettier@v3 we had already installed.

Comment (thanoskrg) [reply package.json:79]: tl;dr: Jest isn't compatible with prettier@3 for formatting inline snapshots.

Comment (jmike) [reply package.json:79]: Yeah, I saw the comment in the code and have read the link. What I didn't understand is why we needed prettier v2 - you explained it ver well below.

> tl;dr: Jest isn't compatible with prettier@3 for formatting inline snapshots.

However, you didn't explain what `npm:` does. Can you elaborate please?

Also, 

> The root cause was that the new tests were failing, because I couldn't get inline snapshots to work with the prettier@v3 we had already installed.

How is it possible that the tests fail because of prettier?

Comment (thanoskrg) [reply package.json:79]: > what npm: does

Actually `npm:` is a sugar syntax, pointing `npm` package manager to understand where to retrieve `prettier-2` from. An alternative of that syntax is the `foo: file:../packages/foo`, where in that case this `file:` sugar syntax points `npm` to install the `foo` package from a relative path.

> How is it possible that the tests fail because of prettier?

Notice that the NEW tests were failing, not all tests. And that's because `jest` is using `prettier` in order to format the inline snapshots, and in our case because of the incompatibility of the `prettier` version it couldn't write the NEW snapshots. The old tests were normally passing.

Screenshot 2023-10-27 at 9 38 35 AM

Comment (jmike) [reply package.json:79]: Re: `npm:`, couldn't this be written as...

```
"prettier-2": "prettier@^2.8.8",
```

I am just wondering whether `npm:` will be introducing a discrepancy with other modules. 

Finally, should we maybe just drop stupid jest integration with prettier?

**Review by jmike — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (thanoskrg) [reply package.json]: Rolled back to prettier@v2 to make sure that we don't introduce two different versions of this dev dependency just for the `jest` compatibility.
